File: 1/solve.py
with open('input', 'r') as fh:
    numbers = fh.readlines()

numbers = [int(n.strip()) for n in numbers]
print(numbers)

print('Part 1')
for i1, n1 in enumerate(numbers):
    for n2 in numbers[i1:]:
        if n1 + n2 == 2020:
            print(n1, n2, n1*n2)

print('Part 2')
for i1, n1 in enumerate(numbers):
    for i2, n2 in enumerate(numbers[i1:]):
        for n3 in numbers[i1+i2:]:
            if n1 + n2 + n3 == 2020:
                print(n1, n2, n3, n1*n2*n3)

File: 1/solve.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("inut")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	var numbers []int
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		//fmt.Println(line)
		n, _ := strconv.Atoi(line)
		numbers = append(numbers, n)
	}
	//fmt.Println(numbers)

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	fmt.Println("Part 1")
	for i1, num1 := range numbers {
		for _, num2 := range numbers[i1:] {
			if num1+num2 == 2020 {
				fmt.Println(num1, num2, num1*num2)
			}
		}
	}

	fmt.Println("Part 2")
	for i1, num1 := range numbers {
		for i2, num2 := range numbers[i1:] {
			for _, num3 := range numbers[i1+i2:] {
				if num1+num2+num3 == 2020 {
					fmt.Println(num1, num2, num3, num1*num2*num3)
				}
			}
		}
	}
}
